### PR TITLE
add new issue templates and update pull request template

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
sync the pull request template that we are using across several repositories and add issue templates that we also used in some repositories

using this if the repository does not have those it will use the ones defined in this repo.

#### Ticket Link
n/a

### CODE REVIEW

**Review by amyblais — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -19,3 +19,35 @@ If this pull request addresses a Help Wanted ticket, please link the relevant Gi
...
+Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
+```
+
+```
+NONE
+```
+-->
+```release-note
+
+```
```
Comment (hanzei) [inline PULL_REQUEST_TEMPLATE.md:53]: Not every repo is using PR descriptions to fetch the release notes e.g. plugins don't. Should this part of the template only be used by repo that use the release notes automation?

File: .github/ISSUE_TEMPLATE/troubleshooting-question.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+name: Troubleshooting question
+about: Ask a question to solve installation and configuration issues
+
+---
+
+For troubleshooting, see [https://www.mattermost.org/troubleshoot/](https://www.mattermost.org/troubleshoot/).
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/troubleshooting-question.md:7]: Troubleshooting for plugins mostly happens in GitHub issues. To which repos does the troubleshooting steps apply?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+For feature proposals see: http://www.mattermost.org/feature-requests/
+
+If you've found a bug--something appears unintentional--please follow these steps:
+
+1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) to ensure that the ticket does not already exist.
+2. Confirm your issue does not involve security. Otherwise, please see our [Responsible Disclosure Policy](https://about.mattermost.com/report-security-issue/).
+3. [File a new issue](https://github.com/mattermost/mattermost-server/issues/new) using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed.
+
+#### Summary
+Bug report in one concise sentence
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/bug_report.md:21]: Should this be HTML comments to be make the template easier to use?

File: .github/ISSUE_TEMPLATE/troubleshooting-question.md (hunk 2 above)
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/troubleshooting-question.md:7]: It seems weird to have a issue template if we don't want community members to use it. Should we use [custom links](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) and link directly instead?

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+
+---
+
+Suggest features in the Mattermost feature idea forum so it can be discussed, upvoted and considered for a [help wanted ticket](https://docs.mattermost.com/process/help-wanted.html).
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/feature_request.md:7]: Same here. For plugins, suggestions should to into GitHub. Should this template be limited to Mobile and Server?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+Per Mattermost guidelines, GitHub issues are for bug reports: <http://www.mattermost.org/filing-issues/>.
+
+For troubleshooting see: http://forum.mattermost.org/.
+For feature proposals see: http://www.mattermost.org/feature-requests/
+
+If you've found a bug--something appears unintentional--please follow these steps:
+
+1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) to ensure that the ticket does not already exist.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/bug_report.md:16]: Is that correct for all repos? E.g. Focalboard uses Focalboard for bugs and plugins use GitHub.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+2. Confirm your issue does not involve security. Otherwise, please see our [Responsible Disclosure Policy](https://about.mattermost.com/report-security-issue/).
+3. [File a new issue](https://github.com/mattermost/mattermost-server/issues/new) using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]: Is this the correct link?

**Comment by jasonblais** [comment]
@cpanato What issue or challenge are we aiming to solve with this change? I share a similar sentiment with Ben, where the templates may not apply to all repositories, e.g. docs vs plugins vs focalboard vs server.

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cpanato) [reply PULL_REQUEST_TEMPLATE.md:53]: we can remove this, but that does not hurt if the repo does not have the bot configured

File: .github/ISSUE_TEMPLATE/troubleshooting-question.md (hunk 2 above)
Comment (cpanato) [reply .github/ISSUE_TEMPLATE/troubleshooting-question.md:7]: we can remove

**Comment by cpanato** [comment]
@jasonblais @hanzei the main idea is to use those when a repository does not have any template in place, then we have at least some basics ones.

we can remove templates that does not make sense

**Comment by jasonblais** [comment]
@cpanato Got it. Does it override a template if it already exists, or only adds it to a repository that doesn't yet have one?

**Comment by cpanato** [comment]
only  if the repo does not have the ones setup here

**Comment by cpanato** [comment]
will close for now

**Pull request closed without merging**